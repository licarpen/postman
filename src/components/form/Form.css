.Form {
  grid-area: form;
  margin: auto;
  text-align: center;
  padding: 0.5vw;
  width: 90%;

  .url{
    display: block;
    width: 100%;
    height: 2vw;
    font-size: 1.5vw;
    margin-bottom: 2vw;

  }

  input[type="radio"] {
    padding: 0.5vw;;
    margin-left: 1vw;
    width: 1vw;
    height: 1vw;
  }

  label{
    padding: 0.5vw;
    margin: 0.5vw;
    width: 5vw;
    height: 2.5vw;
    background: lightgray;
    font-size: 1vw;
    font-family: Arial, Helvetica, sans-serif;
    border-radius: 3em;
  }

  .body{
    display: block;
    margin-top: 20px;
    width: 100%;
    height: 30vw;
    border: lightslategrey 2px solid;
    background: rgb(246, 245, 245);
    overflow: scroll;
  }

  button{
    margin: 20px;

  }

  @media screen and (max-width: 1200px){ 
    .url{
      height: 3vw;
      font-size: 2vw;
    }
  
    input[type="radio"] {
      padding: 0.5vw;;
      margin-left: 2vw;
      width: 3vw;
      height: 3vw;
    }
  
    label{
      padding: 0.5vw;
      margin: 0.5vw;
      width: 8vw;
      height: 5vw;
      background: lightgray;
      font-size: 2vw;
      font-family: Arial, Helvetica, sans-serif;
    }

    .body{
      margin-top: 3vw;
    }
  }
}