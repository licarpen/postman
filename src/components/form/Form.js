import React from 'react';
import PropTypes from 'prop-types';
import styles from './Form.css';

const Form = ({ onSubmit, onChange, url, body, method }) => (
  <form onSubmit={onSubmit} className={styles.Form}>
    <input className={styles.url} type="text" name="url" value={url} placeholder="url" onChange={onChange} />
    <input type="radio" name="method" value="GET" checked={method === 'GET'} onChange={onChange} /><label htmlFor="GET">GET</label>
    <input type="radio" name="method" value="POST" checked={method === 'POST'} onChange={onChange} /><label htmlFor="POST">POST</label>
    <input type="radio" name="method" value="PUT" checked={method === 'PUT'} onChange={onChange} /><label htmlFor="PUT">PUT</label>
    <input type="radio" name="method" value="PATCH" checked={method === 'PATCH'} onChange={onChange} /><label htmlFor="PATCH">PATCH</label>
    <input type="radio" name="method" value="DELETE" checked={method === 'DELETE'} onChange={onChange} /><label htmlFor="DELETE">DELETE</label>
    <input className={styles.body} type="text" name="body" value={body} placeholder="body" onChange={onChange} />
    <button>RESULT</button>
  </form>
);

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired
};

export default Form;
