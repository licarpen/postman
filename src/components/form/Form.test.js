import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Form onChange={() => {}} onSubmit={() => {}} method="POST" url="mywebsite.com" body="text body to send"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
