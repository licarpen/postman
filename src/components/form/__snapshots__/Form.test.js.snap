// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component matches a snapshot 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <input
    className="url"
    name="url"
    onChange={[Function]}
    placeholder="url"
    type="text"
    value="mywebsite.com"
  />
  <input
    checked={false}
    name="method"
    onChange={[Function]}
    type="radio"
    value="GET"
  />
  <label
    htmlFor="GET"
  >
    GET
  </label>
  <input
    checked={true}
    name="method"
    onChange={[Function]}
    type="radio"
    value="POST"
  />
  <label
    htmlFor="POST"
  >
    POST
  </label>
  <input
    checked={false}
    name="method"
    onChange={[Function]}
    type="radio"
    value="PUT"
  />
  <label
    htmlFor="PUT"
  >
    PUT
  </label>
  <input
    checked={false}
    name="method"
    onChange={[Function]}
    type="radio"
    value="PATCH"
  />
  <label
    htmlFor="PATCH"
  >
    PATCH
  </label>
  <input
    checked={false}
    name="method"
    onChange={[Function]}
    type="radio"
    value="DELETE"
  />
  <label
    htmlFor="DELETE"
  >
    DELETE
  </label>
  <input
    className="body"
    name="body"
    onChange={[Function]}
    placeholder="REQUEST BODY (JSON)"
    type="text"
    value="text body to send"
  />
  <button>
    GET RESPONSE
  </button>
</form>
`;
